diff --git a/dist/eslintrc.cjs b/dist/eslintrc.cjs
index 1fa743c91e1420d68aadbdb31d76a51e401e0661..46c8acec3ac29781aa94a74af83eeba765993454 100644
--- a/dist/eslintrc.cjs
+++ b/dist/eslintrc.cjs
@@ -594,17 +594,6 @@ function mergePlugins(target, source) {
                 throw sourceValue.error;
             }
             target[key] = sourceValue;
-        } else if (sourceValue.filePath !== targetValue.filePath) {
-            throw new PluginConflictError(key, [
-                {
-                    filePath: targetValue.filePath,
-                    importerName: targetValue.importerName
-                },
-                {
-                    filePath: sourceValue.filePath,
-                    importerName: sourceValue.importerName
-                }
-            ]);
         }
     }
 }
