diff --git a/index.js b/index.js
index b5eb1205c318cdc6861a56329e31296b898dc952..bdc3ba579130450ceb64041170fc189c490c5be1 100644
--- a/index.js
+++ b/index.js
@@ -1,8 +1,10 @@
 import { createRequire } from 'node:module';
 import { pathToFileURL } from 'node:url';
+import { dirname, join } from 'node:path';
 import { chromium } from 'playwright-core';
-const require = createRequire(import.meta.url);
-const html = String(new URL('index.html', import.meta.url));
+
+const require = createRequire(String(pathToFileURL(__filename)));
+const html = String(pathToFileURL(join(dirname(require.resolve('mermaid-isomorphic')), 'index.html')));
 const mermaidScript = { path: require.resolve('mermaid/dist/mermaid.js') };
 const faStyle = {
     // We use url, not path. If we use path, the fonts can’t be resolved.