diff --git a/index.js b/index.js
index cc1556c6cefd1c147d00f197ea0f02ed0dbefc69..8b4c25545dd9d866225ecee7b588acfdc7215f8e 100644
--- a/index.js
+++ b/index.js
@@ -122,7 +122,7 @@ function createPanZoom(domElement, options) {
     dispose: dispose,
     moveBy: internalMoveBy,
     moveTo: moveTo,
-    smoothMoveTo: smoothMoveTo, 
+    smoothMoveTo: smoothMoveTo,
     centerOn: centerOn,
     zoomTo: publicZoomTo,
     zoomAbs: zoomAbs,
@@ -150,7 +150,7 @@ function createPanZoom(domElement, options) {
   };
 
   eventify(api);
-  
+
   var initialX = typeof options.initialX === 'number' ? options.initialX : transform.x;
   var initialY = typeof options.initialY === 'number' ? options.initialY : transform.y;
   var initialZoom = typeof options.initialZoom === 'number' ? options.initialZoom : transform.scale;
@@ -593,13 +593,8 @@ function createPanZoom(domElement, options) {
     beforeTouch(e);
     clearPendingClickEventTimeout();
 
-    if (e.touches.length === 1) {
+    if (e.touches.length === 2) {
       return handleSingleFingerTouch(e, e.touches[0]);
-    } else if (e.touches.length === 2) {
-      // handleTouchMove() will care about pinch zoom.
-      pinchZoomLength = getPinchZoomLength(e.touches[0], e.touches[1]);
-      multiTouch = true;
-      startTouchListenerIfNeeded();
     }
   }
 
@@ -1098,4 +1093,3 @@ function autoRun() {
 }
 
 autoRun();
-	
